packages/mollie-integration/src/mollie-payment-strategy.ts

```diff
@@ -98,56 +101,43 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
             mollie.form &&
             this.shouldShowTSVHostedForm(methodId, gatewayId)
         ) {
-            this._hostedForm = await this._mountCardVerificationfields(mollie.form);
+            this.hostedForm = await this.mountCardVerificationfields(mollie.form);
         } else if (this.isCreditCard(methodId)) {
-            this._mollieClient = await this._loadMollieJs(
+            this.mollieClient = await this._loadMollieJs(
                 merchantId,
                 storeConfig.storeProfile.storeLanguage,
                 testMode,
             );
             this._mountElements();
         }
 
-        this._unsubscribe = this._store.subscribe(
-            async (state) => {
-                if (
-                    state.paymentStrategies.isInitialized({
-                        methodId: options.methodId,
-                        gatewayId: options.gatewayId,
-                    })
-                ) {
-                    const element = document.getElementById(`${gatewayId}-${methodId}-paragraph`);
-
-                    if (element) {
-                        element.remove();
-                    }
-
-                    mollie.disableButton(false);
+        this.unsubscribe = () => {
+            if (
+                this.paymentIntegrationService.getState().isPaymentMethodInitialized({
```

Comment:
Do this method make the same as before? I don't really know what `this._store.subscribe` method used for, however I can't be sure, that new implementation is working in the same way

Reply:
From what I tested current implementation works as the previous code did. I didn't use variable for state because this method is later executed in class method.

Reply:
Cool. Thanks